twisted/internet/inotify.py

```diff
@@ -137,6 +137,7 @@ def _notify(self, filepath, events):
         """
         Callback function used by L{INotify} to dispatch an event.
         """
+        filepath = filepath.asBytesMode()
```

Comment:
ok... so FilePath with both Unicode and Binary mode does not help to much here.

We can update the docstring and inform users that the notified paths are in byte mode and is their responsability to decode the paths.

So yes, if we don't document it, it is surprising :) 

Otherwise it should be fine. A lot of py3 porting was to move from str to bytes when encoding is not well defined.. so I think that notifying bytes in py3 is ok.

For the long run, I would introduce a BytesFilePath which is FilePath.asBytesMode() only and INotify will only work with such a path... in this way the passed values have self documented types.